drivers/usb/device/usb_dc_sam.c

```diff
@@ -20,6 +20,29 @@ LOG_MODULE_REGISTER(usb_dc_sam);
 #define USBHS_RAM_ADDR		(0xA0100000)
 #endif
 
+/*
+ * The new Atmel DFP headers provide mode-specific interrupt register field
+ * definitions.  Map the existing generic definitions to these.
+ */
+#ifndef USBHS_DEVEPTISR_CTRL_RXSTPI
+#define USBHS_DEVEPTISR_CTRL_RXSTPI	USBHS_DEVEPTISR_RXSTPI
+#endif
+#ifndef USBHS_DEVEPTICR_CTRL_RXSTPIC
+#define USBHS_DEVEPTICR_CTRL_RXSTPIC	USBHS_DEVEPTICR_RXSTPIC
+#endif
+#ifndef USBHS_DEVEPTIMR_CTRL_STALLRQ
+#define USBHS_DEVEPTIMR_CTRL_STALLRQ	USBHS_DEVEPTIMR_STALLRQ
+#endif
+#ifndef USBHS_DEVEPTIER_CTRL_RXSTPES
+#define USBHS_DEVEPTIER_CTRL_RXSTPES	USBHS_DEVEPTIER_RXSTPES
+#endif
+#ifndef USBHS_DEVEPTIER_CTRL_STALLRQS
+#define USBHS_DEVEPTIER_CTRL_STALLRQS	USBHS_DEVEPTIER_STALLRQS
+#endif
+#ifndef USBHS_DEVEPTIDR_CTRL_STALLRQC
+#define USBHS_DEVEPTIDR_CTRL_STALLRQC	USBHS_DEVEPTIDR_STALLRQC
+#endif
```

Comment:
Atmel and now Microchip has been always pretty bad in maintaining their header files. It's not the first time we have an issue like this one. However, so far we've been always patching the header files rather than the drivers. It makes it easier to maintain out of tree code since at times Atmel does fix these errors. Once their header files are consistent we can simply drop the patch.

Reply:
This change, unlike the pointless changes like [`XDMAC_CHID` to `XdmacChid`](https://github.com/zephyrproject-rtos/hal_atmel/pull/8#discussion_r391870983), seems to be both intentional and well meaning (they are taking a more verbose approach and making the interrupt register field values to be defined per endpoint type, which is definitely a good thing and, hopefully, the direction they are moving towards).

Reply:
Sure, this change goes the right direction. As you mentioned in the commit message
"as the revision A variant headers (e.g. same70a) in the latest DFPs, at the time of writing, still provide only the generic definitions."
Microchip messed up again since they've made the change in B variants only. That made header files inconsistent. Hopefully they will fix it in the future. So far we've been fixing such problems by patching the header files.

I'm not insisting, didn't mark this as blocking.